offline/cable_input.F90

```diff
@@ -2887,5 +2924,377 @@ SUBROUTINE allocate_cable_vars(air,bgc,canopy,met,bal,                         &
+
+  ! Iterate through our possible variable names to find the relevant name
+  FindVar: DO VarNameIter = 1, SIZE(Dataset%VarNames)
+    ok = NF90_INQ_VARID(Dataset%CurrentFileID, Dataset%VarNames(VarNameIter),&
+      Dataset%CurrentVarID)
+    IF (ok == NF90_NOERR) THEN
+      EXIT FindVar
+    END IF
+  END DO FindVar
+
+  ! Handle the case where we don't find the variable
+  IF (ok /= NF90_NOERR) THEN
+    CALL handle_err(ok, "Error finding desired variable in "//Dataset%FileNames&
+      (Dataset%CurrentFileIndx))
+  END IF
+
+END SUBROUTINE find_variable_ID
+
+!==============================================================================
+!
+! Name: read_metvals
+!
+! Purpose: Read a record from a SPATIO_TEMPORAL_DATASET from the year and
+!         day specified into the provided DataArr provided using the
+!         landmask (specified with LandIDx, LandIDy).
+!
+! Called from: cable_cru_TRENDY.F90
+!
+! Calls: find_variable_ID
+!
 !==============================================================================
+
+SUBROUTINE read_metvals(STD, DataArr, LandIDx, LandIDy, Year, DayOfYear,&
+    LeapYears, xDimSize, yDimSize, DirectRead)
+  !*## Purpose
+  !
+  ! Read the met forcing values for a given date into the met data array.
+  !
+  !## Method
+  !
+  ! Uses the ```Year``` and ```DayOfYear``` to read a specific record from the
+  ! ```SPATIO_TEMPORAL_DATASET```. Reads from the 2D met data into a 1D array
+  ! using the point IDs in ```LandIDx``` and ```LandIDy```.
+
+  ! Get the data from a day
+  TYPE(SPATIO_TEMPORAL_DATASET), INTENT(INOUT)  :: STD
+  REAL, DIMENSION(:), ALLOCATABLE, INTENT(INOUT)    :: DataArr
+  INTEGER, DIMENSION(:), POINTER, INTENT(IN)    :: LandIDx, LandIDy
+
+  INTEGER, INTENT(IN) :: DayOfYear, Year
+  LOGICAL, INTENT(IN) :: LeapYears, DirectRead
+  
+  ! Size of the global data array
+  INTEGER, INTENT(IN) :: xDimSize, yDimSize
+
+  ! We'll need to compute the record index to grab
+  INTEGER     :: YearIndex, TimeIndex
+
+  ! Loop Iterators
+  INTEGER     :: FileIndx, VarNameIndx, YearIter, LandCell
+
+  ! Status checker
+  INTEGER  :: ok
+
+  ! Temporary array to store the data read from file
+  REAL, DIMENSION(:, :), ALLOCATABLE  :: TmpArray
+
+  TimeIndex = DayOfYear
+  YearIndex = Year
+
+  ! We've already opened a file, check whether the current year is open
+  IF ((Year >= STD%StartYear(STD%CurrentFileIndx)) .AND.&
+    (Year <= STD%EndYear(STD%CurrentFileIndx))) THEN
+    ! In this instance, we don't need to do anything
+    CONTINUE
+  ELSE
+    ok = NF90_CLOSE(STD%CurrentFileID)
+
+    ! If the requested is:
+    !   - Before the time-range of our data, then use the first day from the
+    !     first file in the dataset
+    !   - After the time-range of our data, then use the last day from the last
+    !     file in the datasetc
+    IF (YEAR < STD%StartYear(1)) THEN
+      ! Before the first year
+      STD%CurrentFileIndx = 1
+      YearIndex = STD%StartYear(1)
+      TimeIndex = 1
```

Comment:
This was to get bit-wise reproducibility with the existing routines. I couldn't tell you the why unfortunately. I suspect it might be to handle the `prevTmax` and `nextTmin` "consistently" at the start of the centennial run, but I agree it would make more sense to get the equivalent day from the nearest year. There's already special handling for these next/prev variables.